docker-files-for-Gitlab-CI-rust/substrate-linux-ci/Dockerfile

```diff
@@ -11,25 +14,35 @@ LABEL maintainer="devops-team@parity.io" \
 
 WORKDIR /builds/parity/substrate/
 
+# rustup directory
+ENV RUSTUP_HOME=/usr/local/rustup \
+	CARGO_HOME=/usr/local/cargo \
+	PATH=/usr/local/cargo/bin:$PATH
+
 # install tools and dependencies
 RUN set -eux; \
 	apt-get -y update; \
 	apt-get install -y --no-install-recommends \
-		software-properties-common curl git \
-		make cmake g++ clang rhash \
-		pkg-config libudev-dev time \
-		libssl-dev; \
+		wget git ca-certificates gcc g++ clang \
+		pkg-config libssl-dev cmake libc6-dev; \
+# needed: wget ca-certificates gcc g++ libssl-dev pkg-config git cmake clang
+# not needed: software-properties-common curl make rhash libudev-dev time
```

Comment:
@tomaka @bkchr is it okay to remove `make rhash libudev-dev`? They didn't show up during build and test.

Reply:
I don't know what `rhash` is, but yeah we're not using `libudev-dev` in Substrate.

Reply:
AFAIR, we use `rhash` to calculate hashes of the binaries we've built. It's just a faster version of  `sha256sum`, and is called from our release scripts. I think it should stay.

`libudev-dev` is used when we build, say, Parity Ethereum with hardware wallets support — but I'm not sure if Substrate supports hardware wallets now, or is it planned short-term

Reply:
This is why I asked.
We use `rhash` to sign releases in Parity Ethereum, and `sha256sum` in Substrate.
Should I change the sign to `rhash`?

Reply:
IMO `sha256sum` is most common and most people already have it installed, so if there is no important reason I suggest to stick to that. A speed-up is not mentioned on their [github page](https://github.com/rhash/RHash).

Reply:
Okay it's faster :)
```
% cat grml64-full_2018.12.iso > /dev/null
% time sha256sum grml64-full_2018.12.iso
33617aba115ccb6f99bd59406ebe04ffdc0b9ac15a997c0bd3800b8ad6c7dc91  grml64-full_2018.12.iso
sha256sum grml64-full_2018.12.iso  2.90s user 0.11s system 99% cpu 3.006 total
% time rhash --sha256 grml64-full_2018.12.iso
33617aba115ccb6f99bd59406ebe04ffdc0b9ac15a997c0bd3800b8ad6c7dc91  grml64-full_2018.12.iso
rhash --sha256 grml64-full_2018.12.iso  1.49s user 0.12s system 99% cpu 1.610 total
% time sha256sum grml64-full_2018.12.iso
33617aba115ccb6f99bd59406ebe04ffdc0b9ac15a997c0bd3800b8ad6c7dc91  grml64-full_2018.12.iso
sha256sum grml64-full_2018.12.iso  2.84s user 0.10s system 99% cpu 2.951 total
```

Reply:
it makes sense then to start using it in Substrate as well.